From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Tue, 10 Sep 2019 13:13:14 +0100
Subject: js/ui/background.js: Match theme login screen color

This way the login animation will appear to expand over the login
screen (system background) instead of suddenly replacing it.

Original author: Didier Roche <didrocks@ubuntu.com>
Original author: Daniel van Vugt <daniel.van.vugt@canonical.com>
Author: Marco Trevisan <marco.trevisan@canonical.com>

Last-Update: 2022-08-30
Forwarded: not-needed
---
 js/ui/background.js | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/js/ui/background.js b/js/ui/background.js
index a68f233..3e44923 100644
--- a/js/ui/background.js
+++ b/js/ui/background.js
@@ -529,7 +529,16 @@ var SystemBackground = GObject.registerClass({
     _init() {
         if (_systemBackground == null) {
             _systemBackground = new Meta.Background({ meta_display: global.display });
-            _systemBackground.set_color(DEFAULT_BACKGROUND_COLOR);
+
+            let backgroundColor = DEFAULT_BACKGROUND_COLOR;
+            if (imports.misc.desktop.is("ubuntu")) {
+                const dummyBgActor = new imports.gi.St.Widget({ name: 'lockDialogGroup' });
+                Main.uiGroup.add_actor(dummyBgActor);
+                backgroundColor = dummyBgActor.get_theme_node().get_background_color();
+                dummyBgActor.destroy();
+            }
+
+            _systemBackground.set_color(backgroundColor);
         }
 
         super._init({
